### PR TITLE
Add improvements to run kue from a worker process

### PR DESCRIPTION
**Pull request by jesucarr**
See issue #574 for more details.

The shutdown doesn't close the process, but it is unrelated to the feature added. I know it is still work in progress so I guess it will be fixed by the time it is released.

### CODE REVIEW

**Comment by behrad** [comment]
shutdown should close the process when `disableSearch` is true

**Comment by jesucarr** [comment]
If you look at the worker examples, disableSearch is true, but doesn't close the process, and anyway it should be true by default as far as I understand... 

I guess this is still a work in progress on your side, since it is unrelated to the code I added.

**Comment by behrad** [comment]
I remember we had a similiar bug which had been fixed, but when search indexes are enabled, the `reds` module is opening new connections to redis and Kue is not able to close them :( Search indexes are turned off by default in 0.9 #412 

Would you break down this PR into two: 
1) only the changes to kue.js as a promotion event addition, 
2) a failing test case for the shutdown problem added to tests

**Comment by jesucarr** [comment]
The PR I sent is exactly your point 1)

Regarding 2) I'm not sure how to create that test case. It seems that the redis client is ended, but the process doesn't exit

File: examples/worker-process-jobs.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      });
+    });
+  });
+};
+
+queue.on('job complete', function(result) {
+  console.log('Job completed with result', result);
+  closeIfCan();
+}).on('promotion:end', function() {
```
Comment (behrad) [inline examples/worker-process-jobs.js]: You cannot depend on `promotion:end` being called in every node in a clustered deployment.
Should we add a mock `promotion:end` event to all nodes that are silent (losers to lock race)? or re-view the whole stuff again?

**Comment by jesucarr** [comment]
I had to go ahead with another alternative, and I couldn't have a deep look on how this really works. So yeah feel free to close this PR and review the whole stuff again

**Comment by behrad** [comment]
then closing this for now :)

**Pull request closed without merging**